pkg/virt-handler/vm.go

```diff
@@ -636,6 +638,21 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 	if err != nil {
 		return err
 	}
+
+	detector := isolation.NewSocketBasedIsolationDetector(d.virtShareDir)
+	podEnv, err := detector.Detect(vmi)
+
+	if err != nil {
+		return err
+	}
+	nodeEnv := isolation.NodeIsolationResult()
+
+	// Here we would loop over registered pod modifiers
+	err = devices.NewKVM().Setup(nodeEnv, podEnv)
```

Comment:
are there any issues with this function getting called multiple times for the same VMI?

Reply:
Like it is right now yes. We will have to provide here something anyway, because we are for instance also fetching secrets on each sync run which does definitely not scale.